### PR TITLE
fix notification translation after add to cart

### LINKED ISSUE #3716: Add to cart notification is in the wrong language
## Current behavior

When you add an product in the german store to the cart it shows an english message in the notification. 

image

## Expected behavior

Show german text in the notification. 

## Steps to reproduce the issue

1. got to here https://next.storefrontcloud.io/de/men/tops-men/tees-men/tees-16/balboa-persistence-tee-531.html?childSku=MS03-XS-Gray
2. Add the product to cart
3. See english message in the top right corner

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [x] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Chrome 77.*
- OS: 
- Node: 
- Code Version: 1.11 Release branch

## Additional information

When you look into the core/modules/cart/helpers/notifications.ts file here the notifications are created and at the point of creation the language is english so the notification is english and not german.

### PR DESCRIPTION
**Pull request by gibkigonzo**
### Related issues

closes #3716

### Short description and why it's useful

It fixes issue, but I think much better solution would be if we remove all direct `i18n` imports and make some stateless plugin/service which will always take fresh instance of `i18n`. This plugin will create message objects. :thinking: 

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by andrzejewsky** [comment]
@gibkigonzo Basically notifications are a theme thing. This is the component that is really related and depends only on installed theme. Our vuex / core logic should return only the data / codes that could be handled by theme and then that theme would decide whether it wants to do something with that fact (something happened) or not. Of course don't refactor this now, but keep in mind about that for the future.

File: core/i18n/resource/i18n/pt-PT.csv [hunk 1]
```diff
@@ -22,6 +22,7 @@
 "Proceed to checkout","Ir para Finalizar Compra"
 "Product has been added to the cart!","O produto foi adicionado ao Cesto de Compras!"
 "Product price is unknown, product cannot be added to the cart!","O preço do produto é desconhecido, o produto não pode ser adicionado ao Cesto de Compras!"
+"Product quantity has been updated!","Product quantity has been updated!"
```
[the remark this answers is not in the document]
Comment (gibkigonzo) [reply core/i18n/resource/i18n/pt-PT.csv]: ok :) I will remember

File: core/modules/cart/helpers/notifications.ts [hunk 2]
```diff
@@ -3,39 +3,39 @@ import { currentStoreView, localizedRoute } from '@vue-storefront/core/lib/multi
 import i18n from '@vue-storefront/i18n';
 import config from 'config';
 
-const proceedToCheckoutAction = {
+const proceedToCheckoutAction = () => ({
```
[the remark this answers is not in the document]
Comment (gibkigonzo) [reply core/modules/cart/helpers/notifications.ts]: Good point, I found more to fix in `core/modules/order/store/actions.ts`

**Comment by gibkigonzo** [comment]
@pkarw @andrzejewsky I've updated pr. This change https://github.com/DivanteLtd/vue-storefront/pull/3724/commits/b921d0ef7d757b2ed5307c9b16f077bfa714fe9a resolves problem from issue https://github.com/DivanteLtd/vue-storefront/issues/3716#issuecomment-543153445 We already wait for language to load in prepareStoreView, so this loading is unnecessary. It's async so it can load after one from prepareStoreView

**Comment by ArturDivante** [comment]
Ok after tests :)

**Pull request merged**